### PR TITLE
DOC-8536 initial update of sample-application intro

### PR DESCRIPTION
**Pull request by osfameron**
Each travel-sample app backend app already has good documentation
in its README.md on git, but we need an entrypoint in the SDK docs
too.

This commit:

  * uses Asciidoc attributes to provide a shared skeleton that could be used
  with partials across all the backend apps.
  * mostly defers to the README.
  * has a number of marked TODOs as next steps

*NOTE*: this is a draft, just first commit, to discuss with Maria and Richard!

### CODE REVIEW

**Comment by maria-robobug** [comment]
Nice! 

I like how you've parameterized things so that this document can be more generic, could this live in `sdk-common` somehow?

If it's doable then we would just have to reference `sdk-common` when updating all the other SDK sample app pages 🤔

File: modules/hello-world/pages/sample-application.adoc [hunk 1]
```diff
@@ -5,69 +5,77 @@
...
+== Quick Start
 
-== Preparation
+Fetch the https://github.com/couchbaselabs/{travel-sample-git-project}[Couchbase {name-sdk} travel-sample Application REST Backend] from github:
 
-As well as the xref:start-using-sdk.adoc[Node.js SDK 3.0] and Couchbase Server, 
-set up as described above, you will need `git` to fetch the travel sample application code:
-
-[source,bash]
+[source,bash,subs="+attributes"]
```
Comment (RichardSmedley) [inline modules/hello-world/pages/sample-application.adoc]: Needs to be console,
like in the other docs - 
bash highlighting is geared towards .sh scripts  ;)

File: modules/hello-world/pages/sample-application.adoc [hunk 2]
```diff
@@ -5,69 +5,77 @@
 :page-topic-type: tutorial
 :page-aliases: ROOT:sample-application,ROOT:tutorial4,ROOT:sample-app-backend
 
+:travel-sample-git-project: try-cb-nodejs
+:travel-sample-entrypoint: index.js
+
+// TODO: move these to attributes.adoc
```
Comment (RichardSmedley) [inline modules/hello-world/pages/sample-application.adoc]: :+1:

**Comment by osfameron** [comment]
> I like how you've parameterized things so that this document can be more generic, could this live in `sdk-common` somehow?
Yes! that's the plan, but seemed clearer to keep in one repo for discussion for first cut, to see if we want to do it.

**Review by RichardSmedley — COMMENTED** [review]
LGTM
Will need a script to regularly copy to README (or a Git Action...)
as the other repo doesn't speak Antora.
:+1:

**Comment by RichardSmedley** [comment]
> Will need a script to regularly copy to README (or a Git Action...)
> as the other repo doesn't speak Antora.

That is, if we're going to do that as well :)

**Comment by osfameron** [comment]
> Will need a script to regularly copy to README (or a Git Action...)
> as the other repo doesn't speak Antora.

Ooo, interesting - I think the docs entrypoint is slightly different from the github project README? But that's an option I guess, if we fleshed this out to include the same level of detail.

**Comment by RichardSmedley** [comment]
> Ooo, interesting - I think the docs entrypoint is slightly different from the github project README?
> But that's an option I guess, if we fleshed this out to include the same level of detail.

Just a passing thought, as everything's getting nicely reusable,
but no need to fix the whole universe at once 
*Maybe :D

**Comment by maria-robobug** [comment]
Yeah I would also agree that the Github project README would be slightly different from the doc itself. But we can review that in future for sure.

**Comment by osfameron** [comment]
Requires https://github.com/couchbase/docs-sdk-common/pull/23

**Pull request merged**